src/amt/APFProcessor.ts

```diff
@@ -68,92 +69,124 @@ export enum APFChannelOpenFailureReasonCode {
-        return APFProcessor.channelOpenFailure(socket, len, data)
+        return APFProcessor.channelOpenFailure(socket, data.length, data)
       case APFProtocol.CHANNEL_CLOSE:
-        return APFProcessor.channelClose(socket, len, data)
+        return APFProcessor.channelClose(socket, data.length, data)
       case APFProtocol.CHANNEL_WINDOW_ADJUST:
-        return APFProcessor.channelWindowAdjust(socket, len, data)
+        return APFProcessor.channelWindowAdjust(socket, data.length, data)
       case APFProtocol.CHANNEL_DATA:
-        return APFProcessor.channelData(socket, len, data)
+        return APFProcessor.channelData(socket, data.length, data)
       case APFProtocol.DISCONNECT:
-        return APFProcessor.disconnect(socket, len, data)
+        return APFProcessor.disconnect(socket, data.length, data)
       default: {
         logger.warn(`${messages.MPS_UNKNOWN_CIRA_COMMAND}: ${cmd}`)
         return -1
       }
     }
   },
 
-  disconnect: (socket: CIRASocket, len: number, data: string): number => {
-    if (len < 7) return 0
-    const ReasonCode: number = Common.ReadInt(data, 1)
-    logger.silly(`${messages.MPS_DISCONNECT}, ${ReasonCode.toString()}`)
+  disconnect: (socket: CIRASocket, dataLen: number, data: string): number => {
+    if (dataLen < 7) return 0
+    let parseIndex: number = 1
+    const reasonCode: number = Common.ReadInt(data, parseIndex)
+    parseIndex += 4
+    // reserved short
+    const reserved: number = Common.ReadShort(data, parseIndex)
+    parseIndex += 2
+    if (reserved !== 0) {
+      logger.warn(`${messages.MPS_WARN_RESERVED_VALUE}, ${reserved.toString()}`)
+    }
+    logger.silly(`${messages.MPS_DISCONNECT}, ${reasonCode.toString()}`)
     APFProcessor.APFEvents.emit('disconnected', socket.tag.nodeid)
-    return 7
+    return parseIndex
   },
 
-  channelData: (socket: CIRASocket, len: number, data: string): number => {
-    if (len < 9) return 0
-    const RecipientChannel: number = Common.ReadInt(data, 1)
-    const LengthOfData: number = Common.ReadInt(data, 5)
-    if (len < 9 + LengthOfData) return 0
-    logger.silly(`${messages.MPS_CHANNEL_DATA}, ${RecipientChannel.toString()}, ${LengthOfData.toString()}`)
-    const cirachannel = socket.tag.channels[RecipientChannel]
+  channelData: (socket: CIRASocket, dataLen: number, data: string): number => {
+    if (dataLen < 9) {
+      return 0
+    }
+    let parseIndex: number = 1
+    const recipientChnlId: number = Common.ReadInt(data, parseIndex)
+    parseIndex += 4
+    const chnlDataLen: number = Common.ReadInt(data, parseIndex)
+    parseIndex += 4
+    if (chnlDataLen > APFBulkDataMaxLength) {
+      logger.warn(`${messages.MPS_ERR_MAXLEN} ${APFBulkDataMaxLength} ` +
+        `channelData:chnlDataLen: ${chnlDataLen} `)
+      return -1
+    }
+    if (dataLen < parseIndex + chnlDataLen) {
+      return 0
+    }
+    const chnlData: string = data.substring(parseIndex, parseIndex + chnlDataLen)
+    parseIndex += chnlDataLen
```

Comment:
please spell out the entire word for variables.  Makes the code easier to read.  chnl should be channel.  len should be length, etc.